.github/workflows/workflow.yml

```diff
@@ -186,27 +337,44 @@ jobs:
   # We build a separate job to substitute the halt option.
   # The `deploy` job is dependent on the output of the `check-for-functional-changes`job.
   check-for-functional-changes:
-    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-20.04]
+        numpy: [1.20.3]
+        python: [3.8.10]
+    runs-on: ${{ matrix.os }}
+    name: check-for-functional-changes-${{ matrix.os }}-${{ matrix.numpy }}-${{ matrix.python }}
     if: github.ref == 'refs/heads/master' # Only triggered for the `master` branch
-    needs: [ check-version ]
+    needs: [ test-core, test-country-template, test-extension-template, lint-files, check-version ]
     outputs:
       status: ${{ steps.stop-early.outputs.status }}
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0 # Fetch all the tags
 
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
-          python-version: 3.7.12
+          python-version: ${{ matrix.python }}
 
       - id: stop-early
         run: if "${GITHUB_WORKSPACE}/.github/has-functional-changes.sh" ; then echo "::set-output name=status::success" ; fi # The `check-for-functional-changes` job should always succeed regardless of the `has-functional-changes` script's exit code. Consequently, we do not use that exit code to trigger deploy, but rather a dedicated output variable `status`, to avoid a job failure if the exit code is different from 0. Conversely, if the job fails the entire workflow would be marked as `failed` which is disturbing for contributors.
 
   deploy:
-    runs-on: ubuntu-20.04
+    strategy:
```

Comment:
I see the value of unifying the layout. However, I believe the added complexity is not worth it.

There is a clear difference between the jobs that are run independently from the underlying platform (`deploy` for example) and the tests that depend on the platform setup. I believe the former should be extracted to their own workflow anyway.

I thus suggest to keep the previous version 🙂